blueocean-pipeline-api-impl/src/main/java/io/jenkins/blueocean/preload/FavoritesStatePreloader.java

```diff
@@ -57,30 +60,41 @@
 
     private static final Logger LOGGER = Logger.getLogger(FavoritesStatePreloader.class.getName());
 
+    private static final int DEFAULT_LIMIT = 26;
+
     @Override
     protected FetchData getFetchData(@Nonnull BlueUrlTokenizer blueUrl) {
         User jenkinsUser = User.current();
-        BlueOrganization organization = Iterables.getFirst(OrganizationFactory.getInstance().list(), null);
+        if (jenkinsUser != null) {
+            BlueOrganization organization = Iterables.getFirst(OrganizationFactory.getInstance().list(), null);
+            if (organization != null) {
+                String pipelineFullName = blueUrl.getPart(BlueUrlTokenizer.UrlPart.PIPELINE);
+
+                // don't need this list when at pipeline pages
+                if (pipelineFullName != null) {
+                    return null;
+                }
 
-        if (jenkinsUser != null && organization != null) {
-            UserImpl blueUser = new UserImpl(organization, jenkinsUser);
-            BlueFavoriteContainer favoritesContainer = blueUser.getFavorites();
+                UserImpl blueUser = new UserImpl(organization, jenkinsUser, organization.getUsers());
```

Comment:
@vivek this was the crux of the problem causing the favorites to be re-fetched despite being on the page: URLs didn't match

Reply:
I guess the reason we had it loaded on each page was because favorites were needed to be around on all pages and now we don't need it as we have the new `FavoriteListStatePreloader`?